pymess/backend/dialer/daktela.py

```diff
@@ -0,0 +1,100 @@
+        ('DONE', _l('done'), 'done'),
+        ('RESCHEDULED', _l('rescheduled'), 'rescheduled'),
+    )
+
+    # TODO: clean-up mappings
+    STATES_MAPPING = {
+        STATE.NOT_ASSIGNED: DialerMessage.STATE.UNKNOWN,
+        STATE.READY: DialerMessage.STATE.READY,
+        STATE.RESCHEDULED_BY_DIALER: DialerMessage.STATE.RESCHEDULED,
+        STATE.CALL_IN_PROGRESS: DialerMessage.STATE.IN_PROGRESS,
+        STATE.HANGUP: DialerMessage.STATE.DONE,
+        STATE.DONE: DialerMessage.STATE.DONE,
+        STATE.RESCHEDULED: DialerMessage.STATE.RESCHEDULED,
+    }
+
+    @staticmethod
+    def _get_dialer_api_url(name=None):
+        base_url = '/'.join((
+            settings.DIALER_API_URL,
+            settings.DIALER_API_VERSION,
+            settings.DIALER_API_RESOURCE,
+        ))
+        name = '/' + name if name else ''
+
+        return '{base_url}{name}.json?accessToken={access_token}'.format(
+            base_url=base_url, name=name, access_token=settings.DIALER_API_ACCESS_TOKEN,
+        )
+
+    def _update_dialer_states(self, messages):
+        """
+        Method uses Daktela API to get info about autodialer call status
+        :param messages: list of dialer messages to update
+        """
+        for message in messages:
+            client_url = self._get_dialer_api_url(message.tag)
+            result = generate_session(slug='pymess - Daktela autodialer', related_objects=(message,)).get(client_url)
+
+            try:
+                # TODO: check result params and adjust this method if necessary
+                self.update_message(
+                    message,
+                    state=self.STATES_MAPPING[result.status],
+                    error=result.error_message if result.error_message else None,
+                )
+            except Exception as ex:
+                self.update_message(message, state=DialerMessage.STATE.ERROR, error=force_text(ex))
+
+    def publish_message(self, message):
+        """
+        Method uses Daktela API for sending dialer message
+        :param message: dialer message
+        """
+        client_url = self._get_dialer_api_url()
+        try:
+            payload = {
+                'queue': 'queue number',
+                'number': message.recipient,
+                'customFields': {
+                    'tts_text': message.content,
+                }
+            }
+
+            result = generate_session(slug='pymess - Daktela autodialer', related_objects=(message,)).post(
+                client_url,
+                json=payload,
+            )
+
+            # TODO: check result params and adjust this method if necessary
+            self.update_message(
+                message,
+                state=self.STATES_MAPPING[result.status],
+                error=result.error_message if result.error_message else None,
+                sent_at=timezone.now(),
+                # TODO: is this okay?
+                tag=result.name
```

Comment:
tag je použité pro klienty, který používá pymess může si otagovat zprávy

Reply:
ok, ak teda budem potrebovat niekde ukladat takuto info, asi teda pridam do `extra_data`